src/spikeinterface/core/core_tools.py

```diff
@@ -796,61 +832,84 @@ def recursive_path_modifier(d, func, target="path", copy=True) -> dict:
       * relative/absolute path change
       * docker rebase path change
 
-    Modification is inplace with an optional copy.
+    Modification is in-place with an optional copy.
+    If the dictionary contains extractors, the copy argument is ignored and an exception is raised.
 
     Parameters
     ----------
-    d : dict
-        Extractor dictionary
+    dictionary : dict
+        Extractor dictionary, including "kwargs" key.
     func : function
         Function to apply to the path. It must take a path as input and return a path
     target : str, optional
         String to match to dictionary key, by default 'path'
     copy : bool, optional
         If True the original dictionary is deep copied, by default True (at first call)
+    skip_targets : list or str, optional
+        List of targets to skip
+    skip_warning : bool, optional
+        If True, skip the warning when the dictionary contains extractors, by default False
 
     Returns
     -------
     dict
         Modified dictionary
     """
+    from .base import BaseExtractor
+
+    if skip_targets is None:
+        skip_targets = []
+    elif isinstance(skip_targets, str):
+        skip_targets = [skip_targets]
+
+    if dict_contains_extractors(dictionary):
+        if copy:
+            raise Exception("The copy argument is only available if the input dictionary does not contain objects")
+        else:
+            if not skip_warning:
+                warnings.warn(
+                    "The dictionary contains extractors, so the paths will be modified in-place!" "Use with caution"
+                )
```

Comment:
I am not sure to be agree with inplace modification. I think we should raise error or force a copy always.